perfzero/lib/benchmark.py

```diff
@@ -125,6 +158,8 @@ def run_benchmark(self):
       benchmark_success_results[benchmark_method] = succeeded
       benchmark_output_dirs[benchmark_method] = output_dir
 
+    self._teardown()
```

Comment:
this should be something like:

try:
  run_benchmark_methods()
finally:
  utils.cleanup_tpu(self.config.tpu_parameters)

Reply:
Thanks, changed to move the cleanup_tpu call into a try-finally block.